kuksa-client/kuksa_client/grpc/__init__.py

```diff
@@ -322,11 +322,23 @@ def from_message(cls, message: types_pb2.Datapoint):
         if message.WhichOneof('value') is None:
             logger.warning("No value provided in datapoint!")
             return None
+
+        if message.HasField('timestamp'):
+            # Sanity check, is date before year 2500
+            # Technically up to 9999 is allowed
+            if message.timestamp.seconds > 16725225600:
```

Comment:
why check for year 2500 and not now?

Reply:
@SebastianSchildt - any thoughts here? I.e. shall we have a stricter requirement on times received?

Or is that potentially part of a wider overhaul of how we handle time and timestamps? I am thinking - would it make sense to have a dynamic read-only datapoint like `Kuksa.Databroker.Time` which clients/providers could call to make sure that the client and Databroker has roughly the same understanding of time. Could be useful information for example if token validation fail, or if we would implement a stricter check here which fails as either broker or client does not know the correct time.

Reply:
@erikbosch It might makes sense to put some sanity check here, to make sure the ToDatetime can suceed, but if I look at the finding in the audit, hasn't the fault already happened earlier?  Because I see a proto-related abcktrace

```
File ~/.../site-packages/google/protobuf/internal/well_known_types.py:232, in ,→ Timestamp.ToDatetime(self, tzinfo)
  215 """Converts Timestamp to a datetime.
  ...
  --> 232 delta = datetime.timedelta(
      233     seconds=self.seconds,
      234     microseconds=_RoundTowardZero(self.nanos, _NANOS_PER_MICROSECOND),
235 )
  OverflowError: days=1628906115; must have magnitude <= 999999999}
```

and as this function already receives and ` types_pb2.Datapoint` does it imply the error happened earlier?

Reply:
Ah ignore that, I do see that we call this directly with ` timestamp = message.timestamp.ToDatetime` 

I would just instead of doing an arbitrary limit, call that function in a try/except block, if it fails we can still return none

Reply:
Changed to try/except now, so now we supports dates up to 9999-12-31 🥇